cas-server-core-api/src/main/java/org/jasig/cas/ticket/TicketGrantingTicket.java

```diff
@@ -66,11 +67,12 @@
      * @param service The service for which we are granting a ticket
      * @param expirationPolicy the expiration policy.
      * @param credentialsProvided if the credentials are provided.
+     * @param registeredService the associated registered service
      * @return the service ticket granted to a specific service for the
      * principal of the TicketGrantingTicket
      */
     ServiceTicket grantServiceTicket(String id, Service service,
-        ExpirationPolicy expirationPolicy, boolean credentialsProvided);
+        ExpirationPolicy expirationPolicy, boolean credentialsProvided, RegisteredService registeredService);
```

Comment:
This is what I was referring to that will not work. We should use a simple flag here instead.